src/Components/Components/src/Routing/OnNavigateArgs.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Routing
+{
+    public class OnNavigateArgs
+    {
+        public OnNavigateArgs(string path, CancellationTokenSource cancellationTokenSource)
```

Comment:
I think this should be a `CancellationToken`, not a `CancellationTokenSource`, since the recipient only deals with observing the cancellation state, not setting it.

Reply:
Also, the class/method needs XML docs

Reply:
Agree, I also don't think we need to make this constructor public since we are the only ones invoking this. I'd rather keep construction private so that we don't corner ourselves if we want to change this in the future.

Reply:
> we are the only ones invoking this

Should we account for people writing unit tests?

Reply:
That's a fair point, but then should we just make the properties settable for testing? That's honestly what we do with most of these type of objects in the framework.

I would say we don't need it, and we can add it later if someone asks for it.